### ISSUE TITLE
Potential issue with gatsby-source-strapi

### ISSUE THREAD

**Issue by ChrisKatsaras** [AHgPuK/strapi-mongodb-files#3]
Recently, I've been working on redesigning my frontend using Gatsby. In order to properly integrate Gatsby with Strapi, `gatsby-source-strapi` https://www.gatsbyjs.org/packages/gatsby-source-strapi/ needs to be used. However, I've noticed that the plugin has some issues with pulling data, specifically images when I am using `strapi-mongodb-files`. Some images will simply return null:

![image](https://user-images.githubusercontent.com/11506653/82720202-18424f80-9c7f-11ea-83e8-cd12961fc7b4.png)

However, once I switch the upload plugin provider back to `Local Server` the issue goes away and Gatsby is able to properly pull the data.

I'm curious if anyone else has ran into this issue. @AHgPuK have you by any chance dealt with something similar using `gatsby-source-strapi` and your plugin? 

Any information is greatly appreciated and thanks for the work that you've put in to make this plugin 😃

**Comment by AHgPuK** [comment]
@ChrisKatsaras can you provide me example project source already configured so I can test it?

**Comment by ChrisKatsaras** [comment]
Hi @AHgPuK !
I was able to resolve my issue for the most part by implementing a resolver in my `gatsby-node.js` as suggested in https://github.com/strapi/gatsby-source-strapi/issues/98. 
This implementation works almost perfectly except for files that have spaces in the name or `.` before the extension. I don't think it's an issue in your library as I'm able to pull the file correctly even if it does have spaces or periods. 
The issue happens when the resolver executes during gatsby's build process. I outlined the problem in the issue above and hopefully, someone there can help me. I'll make sure to post it here too if I figure it out.

Thanks for getting back to me!
